### PR TITLE
Feature/issue 4042 vista

### LINKED ISSUE #4042: Re-implement Clock object and indicator in Vue.js
[The existing Clock object](https://github.com/nasa/openmct/blob/master/platform/features/clock/bundle.js#L189) and the [associated indicator](https://github.com/nasa/openmct/blob/master/platform/features/clock/bundle.js#L77) are currently implemented as a legacy bundle and [written in Angular.js](https://github.com/nasa/openmct/blob/master/platform/features/clock/src/controllers/ClockController.js) . They should be re-implemented as a plugin under `/src/plugins` using our current API, with the UI migrated to Vue.

The Timer should not be re-implemented just yet, just the clock object and clock indicator. Re-implementing the timer should be addressed separately as its functionality is a little more nuanced and we need to decide on the correct design approach.

[A followup issue](https://github.com/nasa/openmct/issues/4043) has also been created to enhance the clock object and indicator to support remote clocks, but that can be addressed in a future PR.

Any legacy code associated with the legacy Clock object and indicator should be removed from the code-base.

### PR DESCRIPTION
**Pull request by khalidadil**
Fixes #4042  

### Author Checklist

Check | Passed
-- | --
Changes address original issue? | Y
Unit tests included and/or updated with changes? | In issue #4093 
Command line build passes? | Y
Changes have been smoke-tested? | Y
Testing instructions included? | Y

### Testing instructions
1. If possible, create a clock before the change and then change to the branch with the change. Ensure that the clock continues to display properly and that editing the properties works.
2. Create a clock object and choose some values in the creation dialog. Ensure that the time is displayed and is in the appropriate format based on the options chosen (proper timezone, format, 12/24 hours).
3. Edit the properties to modify the display format and ensure the clock updates with the correct time.
4. Edit the properties to modify the 12/24 hours selection and ensure the clock updates with the correct time.
5. Edit the properties to modify the timezone and ensure the clock updates with the correct time.
6. It would be useful to make sure the timer still works properly as well since code around it was removed for this change.

### CODE REVIEW

File: src/plugins/clock/plugin.js [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+
+            openmct.indicators.add(indicator);
+        }
+
+        openmct.objects.addGetInterceptor({
+            appliesTo: (identifier, domainObject) => {
+                return domainObject && domainObject.type === 'clock';
+            },
+            invoke: (identifier, domainObject) => {
+                if (domainObject
```
Comment (nikhilmandlik) [inline src/plugins/clock/plugin.js:130]: i think we can simply return if there is no `!domainObject.configuration` and if `domainObject` is null or undefined let it throw error.
```
if (!domainObject.configuration) {
     return domainObject;
}
```

File: src/plugins/clock/plugin.js [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+                    const use24 = domainObject.clockFormat[1];
+                    const timezone = domainObject.timezone;
+
+                    domainObject.configuration = {
+                        baseFormat,
+                        use24,
+                        timezone
+                    };
+
+                    openmct.objects.mutate(domainObject, 'configuration', configuration);
```
Comment (nikhilmandlik) [inline src/plugins/clock/plugin.js:144]: not sure about this `, configuration);` cant find variable `configuration`

File: src/plugins/clock/plugin.js [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+            const clockIndicator = new Vue ({
+                components: {
+                    ClockIndicator
+                },
+                provide: {
+                    openmct
+                },
+                data() {
+                    return {
+                        CLOCK_INDICATOR_FORMAT
```
Comment (nikhilmandlik) [inline src/plugins/clock/plugin.js:112]: does this works? or typo?
`indicatorFormat: CLOCK_INDICATOR_FORMAT`

**Review by nikhilmandlik — CHANGES_REQUESTED** [review]
Looks good, some small suggestions/changes.
Also can you add test cases?

File: src/plugins/clock/components/ClockIndicator.vue [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+        });
+    },
+    beforeDestroy() {
+        if (this.unlisten) {
+            this.unlisten();
+        }
+    },
+    methods: {
+        tick(timestamp) {
+            this.timeTextValue = `${moment.utc(timestamp).format(this.indicatorFormat)} UTC`;
```
[the remark this answers is not in the document]
Comment (khalidadil) [reply src/plugins/clock/components/ClockIndicator.vue:61]: yup!

File: src/plugins/clock/components/ClockIndicator.vue [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+    props: {
+        indicatorFormat: {
+            type: String,
+            required: true
+        }
+    },
+    data() {
+        return {
+            timeTextValue: null,
+            unlisten: null
```
[the remark this answers is not in the document]
Comment (khalidadil) [reply src/plugins/clock/components/ClockIndicator.vue:45]: 👍🏽

File: src/plugins/clock/plugin.js (hunk 1 above)
Comment (khalidadil) [reply src/plugins/clock/plugin.js:130]: We'd want to return if there's already a configuration I guess?

File: src/plugins/clock/plugin.js (hunk 2 above)
Comment (khalidadil) [reply src/plugins/clock/plugin.js:144]: yikes, good catch!

File: src/plugins/clock/plugin.js (hunk 3 above)
Comment (khalidadil) [reply src/plugins/clock/plugin.js:112]: this works, but I can rename it to make it clearer

File: src/plugins/clock/components/ClockIndicator.vue (hunk 4 above)
[the remark this answers is not in the document]
Comment (khalidadil) [reply src/plugins/clock/components/ClockIndicator.vue:60]: You can checkout the deleted `src/indicators/ClockIndicator.js` above

**Review by davetsay — APPROVED** [review]

**Pull request merged**